File: Lab_2/task_1/1.py
def compress_string(s):
    compressed = ''
    count = 1

    for i in range(1, len(s) + 1):
        if i < len(s) and s[i] == s[i - 1]:
            count += 1
        else:
            compressed += s[i - 1] + str(count)
            count = 1

    if len(compressed) < len(s):
        return compressed
    else:
        print("Сжатие не уменьшило строку. Возвращена исходная строка.")
        return s

s = input("Введите строку: ")
print("Результат:", compress_string(s))

File: Lab_2/task_1/1.rs
use std::io::{self, Write};

fn compress_string(s: &str) -> String {
    let mut compressed = String::new();
    let mut count = 1;
    let len = s.len();

    for i in 1..=len {
        if i < len && s.as_bytes()[i] == s.as_bytes()[i - 1] {
            count += 1;
        } else {
            compressed.push(s.chars().nth(i - 1).unwrap());
            compressed.push_str(&count.to_string());
            count = 1;
        }
    }

    if compressed.len() < len {
        compressed
    } else {
        println!("Сжатие не уменьшило строку. Возвращена исходная строка.");
        s.to_string()
    }
}

fn main() {
    print!("Введите строку: ");
    io::stdout().flush().unwrap();

    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let input = input.trim();

    println!("Результат: {}", compress_string(input));
}
